src/parcsr_ls/HYPRE_parcsr_ls.h

```diff
@@ -4508,6 +4536,58 @@ HYPRE_ILUSolve( HYPRE_Solver solver,
 HYPRE_Int
 HYPRE_ILUSetMaxIter( HYPRE_Solver solver, HYPRE_Int max_iter );
 
+/**
+ * (Optional) Set the algorithm type for compute the ILU factorization. Options are:
+ *
+ *    -  0 : Non-iterative algorithm (default)
+ *    -  1 : Asynchronous with in-place storage
+ *    -  2 : Asynchronous with explicit storage splitting
+ *    -  3 : Synchronous with explicit storage splitting
+ *    -  4 : Semi-synchronous with explicit storage splitting
+ *
+ * Note: Iterative ILU is available only for zero fill-in and it depends on rocSPARSE.
+ **/
+HYPRE_Int
+HYPRE_ILUSetIterativeSetupType( HYPRE_Solver solver, HYPRE_Int iter_setup_type );
+
+/**
+ * (Optional) Set the compute option for iterative ILU in a bitwise fashion. Options are:
```

Comment:
I would shy away from talking about bits.  Instead, maybe refer to these as "additive options" where several of them can be set at the same time by adding the individual numeric values.  I think you had a good user-level explanation of this somewhere else in the library (having to do with print options?).

Reply:
Yes! I can replicate what I did for `HYPRE_MGRSetPrintLevel` here